src/FINDOLOGIC/Api/Responses/Response.php

```diff
@@ -125,10 +124,24 @@ private static function buildSearchOrNavigationResponse(
     /**
      * Checks if the response is valid. If not, an exception will be thrown.
      *
+     * @param Request $request
      * @param GuzzleResponse $response
      */
-    protected static function checkResponseIsValid($response)
+    protected static function checkResponseIsValid(Request $request, GuzzleResponse $response)
     {
+        if ($request->getOutputAdapter() === OutputAdapter::XML_21) {
+            $xml= new DOMDocument();
+            $xml->loadXML($response->getBody()->getContents());
+
+            libxml_use_internal_errors(true);
+
+            if (!$xml->schemaValidate(__DIR__ . '../../../../../vendor/findologic/xml-response-schema/schema.xsd')) {
```

Comment:
This code only works by coincidence. `__DIR__` will never append a `/` at the end. Drop the first two `..` and you will see that it still works.